### ISSUE TITLE
Recording positional audio

### ISSUE THREAD

**Issue by hasdavid** [mumble-voip/mumble#3574]
Feature request:
When using the record feature, it does not take into account the positional audio.

This is weird, as when using PA, users may not hear other people speaking due to their distance in game. But when listening to the record, everyone's voices are on 100 % volume, resulting in voices talking over each other and the record being unusable.

An idea is to add a check box in the settings, which would allow to record the positional audio, if used. Maybe enable it by default, so that what users hear is what is recorded, to avoid confusion.

**Comment by Kissaki** [comment]
Thinking about recording, my first thought would be being able to hear everyone talk.

I guess with PA enabled is a specific use case. Do you plan on using the recorded audio in combination with recorded video from gameplay? Or do you see use in PA adjusted audio recordings without those? Just trying to understand the use case.

**Comment by hasdavid** [comment]
You're right, thinking about it that way, it probably shouldn't be enabled by default.

And yes, the use case I have in mind is combining the recorded audio with gameplay footage. Mumble's qualities make it a great tool for that. The recording is simple and the ability to produce separate files for each participant allows for further editing.

However, currently the only way to record the positional audio is to use external software. Which is possible, but complicated.

**Comment by Krzmbrzl** [comment]
Without knowing a lot about how recording is implemented, I think that this might be as easy as to move this code block
https://github.com/mumble-voip/mumble/blob/713b92c3c233f8f38200c4a458e0196d3d39a3b0/src/mumble/AudioOutput.cpp#L499-L518
into a function that is either called at the current position (recording all voices without PA) or at the bottom of
https://github.com/mumble-voip/mumble/blob/713b92c3c233f8f38200c4a458e0196d3d39a3b0/src/mumble/AudioOutput.cpp#L520-L551

which should then (in theory) record the positional audio as heard through Mumble.

Add a corresponding UI checkbox somewhere in the settings and be done with it.

Could it really be that simple or am I missing something here?
